OpenSim/Simulation/Model/Ground.cpp

```diff
@@ -41,10 +43,22 @@ using namespace OpenSim;
 */
 Ground::Ground() : PhysicalFrame()
 {
-    setName("ground");
+    setName(GroundNameString);
     setAuthors("Ajay Seth");
 }
 
+/* Ground should never change name */
+void Ground::extendFinalizeFromProperties() {
+    Super::extendFinalizeFromProperties();
+    // Ground is always named "ground"
+    if (getName() != GroundNameString) {
+        std::string msg = getConcreteClassName() + " '" + getName() + "' ";
+        setName(GroundNameString);
+        msg += "was renamed and is being reset to '" + getName() + "'.";
+        std::cout << msg << std::endl;
+    }
+}
+
```

Comment:
Record that this fixes one of the other issues on GitHub.

Reply:
Yep this is point in 3. in  #1876

Reply:
Is component name consistency/checking done across the board in extendFinalizeFromProperties() is this documented?